pcsx2/gui/AppMain.cpp

```diff
@@ -1008,21 +1011,42 @@ void Pcsx2App::OpenGsPanel()
 
 	GdkWindow* draw_window = gtk_widget_get_window(child_window);
 
-#if GTK_MAJOR_VERSION < 3
-	Window Xwindow = GDK_WINDOW_XWINDOW(draw_window);
-#else
-	Window Xwindow = GDK_WINDOW_XID(draw_window);
-#endif
-	Display* XDisplay = GDK_WINDOW_XDISPLAY(draw_window);
+	// For Wayland ShowFullScreen must be called before GetPluginDisplayProperties
+	// so that the viewport's wl_surface is created before we access it.
+	gsFrame->ShowFullScreen( g_Conf->GSWindow.IsFullscreen );
 
-	pDsp[0] = (uptr)XDisplay;
-	pDsp[1] = (uptr)Xwindow;
+#ifdef GDK_WINDOWING_WAYLAND
+	if (GDK_IS_WAYLAND_WINDOW(draw_window))
+	{
+		pDsp[0] = (uptr)gsFrame->GetPluginDisplayPropertiesWaylandEGL();
+		pDsp[1] = (uptr)NULL;
```

Comment:
nullptr here too

Reply:
NULL and nullptr both require a cast to `uptr` since `uptr` is an integer size type (size of a pointer) and not a pointer type.

Reply:
same as https://github.com/PCSX2/pcsx2/pull/4075#discussion_r552731633